bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/guide/GuideProbe.java

```diff
@@ -122,7 +119,9 @@ public Boolean apply(GuideProbe guideProbe) {
 
     /**
      * Indicates the bands that will be used for a given probe.
-     *
+     * Use R-band by default.
      */
-    BandsList getBands();
+    default BandsList getBands() {
+        return RBandsList.instance();
+    }
```

Comment:
It's certainly true that this removes duplicate code, but it also allows you to not think about it when you implement a new guide probe.  If there is no default you're forced to provide an implementation and figure out what it should be.

One of the main drivers for the `default` feature was to allow interfaces that have existing implementations to be extended without requiring every implementation be updated.  We aren't really providing libraries and have control over all the implementations so that doesn't apply to us.

I haven't thought about this much but maybe a good rule would be to only use this feature for convenience methods that you don't expect or want to be overridden in the implementations?

Reply:
Your first paragraph had occurred to me. I have no objections changing it back, which will just take a few minutes anyway. I'll go ahead and do that.